src/Compilers/CSharp/Portable/Lowering/Instrumentation/LocalStateTracingInstrumenter.cs

```diff
@@ -313,7 +313,7 @@ public override void InstrumentBlock(BoundBlock original, LocalRewriter rewriter
                 if (parameterLogger != null)
                 {
                     prologueBuilder.Add(_factory.ExpressionStatement(_factory.Call(receiver: _factory.Local(_scope.ContextVariable), parameterLogger,
-                        MakeStoreLoggerArguments(parameterLogger.Parameters[0], parameter, parameter.Type, _factory.Parameter(parameter), refAssignmentSourceIndex: null, _factory.Literal((ushort)parameter.Ordinal)))));
+                        MakeStoreLoggerArguments(parameterLogger.Parameters[0], parameter, parameter.Type, _factory.Parameter(parameter), refAssignmentSourceIndex: null, _factory.ParameterId(parameter)))));
```

Comment:
:memo: The change to use a `BoundParameterId` instead of the parameter's ordinal caused a bit of churn in `LocalStateTracingTests` because of logic differentiating hoisted from non-hoisted parameters in `EmitParameterIdExpression`. Those test changes are in commit 2.